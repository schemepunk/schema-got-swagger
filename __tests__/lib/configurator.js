'use strict';

const configurator = require('../../lib/configurator');
const _ = require('lodash');
const { SchemaGotSwaggerError } = require('../../lib/SchemaGotSwaggerError');

let tmpMocks = [];

beforeEach(() => {
  tmpMocks.forEach(mock => mock.mockRestore());
  tmpMocks = [];
  jest.resetAllMocks();
  jest.spyOn(Date, 'now').mockReturnValue(2000);
});

afterAll(() => {
  jest.restoreAllMocks();
});

describe('Configurator', () => {
  test('Get yaml files', () => {
    expect.assertions(1);
    return configurator.getSgsDefaults()
    .then(data => expect(data).toMatchSnapshot())
  });
  test('Bad path throws', () => {
    expect.assertions(1);
    return configurator.promiseYamlRetrieval('notThere')
    .catch(e => expect(e).toBeInstanceOf(SchemaGotSwaggerError));
  })
  test('mergeConfig', () => {
    return configurator.mergeConfig('Sgs', {apiType: 'custom'})
     .then(result => expect(result).toMatchSnapshot());
  })
});
