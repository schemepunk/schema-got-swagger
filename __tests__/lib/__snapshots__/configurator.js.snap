// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configurator Get yaml files 1`] = `
Object {
  "apiType": "jsonApi",
  "sgsType": "semver",
  "swaggerVersion": "3.0.0",
}
`;

exports[`Configurator mergeConfig 1`] = `
Object {
  "apiType": "custom",
  "sgsType": "semver",
  "swaggerVersion": "3.0.0",
}
`;
